pages/video-detail/video-detail.controller.js

```diff
@@ -0,0 +1,44 @@
+angular.module('app')
+	.controller('VideoDetailController', ['$stateParams', 'session', 'video', 'user', '$timeout', VideoDetail]);
+
+function VideoDetail($stateParams, session, video, user, $timeout) {
+	var self = this;
+
+	self.id = $stateParams.id;
+	self.code = '';
+	self.loading = true;
+	self.loadingAuthor = true;
+	self.session = session; 
+
+	self.video = video;
+	self.user = user;
+
+	self.showMore = showMore();
+
+	if(self.video && self.video.data && self.video.data.id !== self.id) {
+		self.video.reset();
+	}
+	
+	// ToDo: Check invalid video :S
+	self.video.get(self.id, function() {
+		self.loading = false;
+		self.user.get(self.video.data.owner, function() {
```

Comment:
This hasn't to be tackled now, but from the experience of this cycle working with frontend, for me this kind of pattern of use has some weird smell. I'll explain myself:

* If services are singletons, then they are going to persist in subsequent calls, so they are their data.
* This lead to extrange or not desired behaviours such as one than happened in video donwload page, that showed data from last visited video instead of current one, and extrange workarrounds like videoService.reset()
* It also add implicit dependency of view to service, as view access to Controller.service.data, and if I change the implementation of my service, I would need to change both controller and view
* For me a service would allow me to retrieve some data, but I will store and manage in the controller, passing it back to service if further modifications are needed (such as entity update). Thus, the service will provide me with a repository pattern for the chosen model, and abstracts the controller in how to get and manage the data (as it's done in backend)
* Another use case is to store locally data that should not change, such as (maybe) the curren logged in user. In this case I'll maybe choose a singleton to store (and eventually update) it.

In summary, it makes more sense to me, not to store this kind of data on services singlentons, but instead on controller instances. Controller retrieves somehow (from the service in this case, from local caches, maybe in others) an object and stores locally, and view just know that controller has an object to display.

Reply:
Humm... There's a big topic here, and actually, it depends on so much things. 
 - If you move the data from service to controller, you can't share that data through different controllers, at least in a recommended way. 
 - Also, if you move it to controller, yes, you're fit to service definitions, but you only define the way to manage data in one point of the code instead of doing it several times, one per controller.
 - Keeping the data in services allows you to automatically pre-cache results for several requests (since controllers dies when you get out the view, so it don't keep any previous request)

We can set a day, and do together (cc @DevStarlight) some workarrounds and decide. Might be the same day we talk about async and Ecma rules ;)

Reply:
yep, definitively :)

For me it's all about the two use cases that I mentioned avobe:
* Generally you don't want to share data through different controllers, avoiding extrange behaviour mentioned.
* Sometimes you want, as currently logged in user, then it has sense to have that data in a singleton ;)